### PR TITLE
Webscraping with Javascript

### LINKED ISSUE #3177: Introduction to Web Scraping with JavaScript and Puppeteer
### Introduction to Web Scraping with Javascript and Puppeteer

When developers need data for their web apps, they usually lookup for an API (Application Programming Interface), hoping that it will have the data they need for the app. This API is created by someone else and we expect it to have the exact data that we need. But this is not always the case. We may also need data from a certain website which unfortunately does not have an API. This is where web scraping comes in.

In this article, we are going to learn how we can get data from any website using vanilla JavaScript and Puppeteer. Puppeteer is a Node library that provides a high-level API to control Chrome or Chromium over the DevTools Protocol.

### Prerequisites:
- Basic knowledge of HTML and CSS.
- Intermediate JavaScript knowledge.
- Understanding of arrow functions in JavaScript.
- Understanding of asynchronous JavaScript.

### Key takeaways: 
- How to get data from any website.

### References: 
N/A

### Templates to use as guides
N/A

### PR DESCRIPTION
**Pull request by kimanikevin254**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3177

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by marienjus** [comment]
Hi I'll be your peer reviewer. Glad to see you've included the author details in your first PR!

**Comment by marienjus** [comment]
Remove the backticks and include your issue number as this fixes #3177. I will be taking you through preliminary review

**Comment by kimanikevin254** [comment]
This fixes #3177

**Comment by Kayere** [comment]
Hello @marienjus - Thank you for showing interest in reviewing this article. However, test reviewers are only allowed to review one article at a time. I am going to un-assign you so that another reviewer can review this article.

Regards.

**Comment by hectorkambow** [comment]
@kimanikevin254 
[Webscraping with Javascript #3413.pdf](https://github.com/section-engineering-education/engineering-education/files/7027915/Webscraping.with.Javascript.3413.pdf)


There appear to be many sentences in this article that are taken verbatim from third-party sources. We can't publish any content that is not original or properly cited/referenced. There are many free online tools where you can check this (such as quetext.com, and if you're uncertain about the definitions around plagiarism, this is a [good resource to reference](https://www.scribbr.com/category/plagiarism/).


cc @Dawe-7

**Comment by kimanikevin254** [comment]
@hectorkambow Could you please consider reopening this PR. I fixed the plagiarism mistakes. The problem was that I had included definitions from the official products page.

**Comment by kimanikevin254** [comment]
@hectorkambow ?

**Comment by BrianaNzivu** [comment]
@hectorkambow I have re-opened this PR. 
Maybe you can run a check again to ensure of it is fine.
@Dawe-7 you can resume with the review if everything is alright. 

Regards. 🚀 

Cc @Dawe-7
Cc @keerthivarumbudy

**Comment by Dawe-7** [comment]
> @hectorkambow I have re-opened this PR.
> Maybe you can run a check again to ensure of it is fine.
> @Dawe-7 you can resume with the review if everything is alright.
> 
> Regards. 
> 
> Cc @Dawe-7
> Cc @kimanikevin254

@hectorkambow, Could we kindly have another plagiarism check on this PR before i begin the review.

**Comment by hectorkambow** [comment]
plagiarism check done @kimanikevin254 

cc @Dawe-7

File: content/articles/introduction-to-web-scraping-with-javascript-and-puppeteer/index.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+Since our server is working, it's time we set up puppeteer to get started with web scraping.
+In our **index.js** file, add the following line of code just below the line that sets the port number:
+
+```js
+const puppeteer = require("puppeteer");
+```
+
+The above code imports the puppeteer package into our app.
+
+In this tutorial, this is the URL of the product that we will be tracking on amazon: https://www.amazon.com/Redragon-S101-Keyboard-Ergonomic-Programmable/dp/B00NLZUM36/
```
Comment (Dawe-7) [inline content/articles/introduction-to-web-scraping-with-javascript-and-puppeteer/index.md:109]:
```suggestion
In this tutorial, this is the [URL](https://www.amazon.com/Redragon-S101-Keyboard-Ergonomic-Programmable/dp/B00NLZUM36/)  of the product that we will be tracking on amazon: 
```

File: content/articles/introduction-to-web-scraping-with-javascript-and-puppeteer/index.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+date: 2021-09-05T00:00:00-17:50
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/introduction-to-web-scraping-with-javascript-and-puppeteer/hero.jpg
+    alt: Introduction to Web Scraping With JavaScript and Puppeteer Hero Image
+---
+Most of our web apps require data to process and present to the user. There are many sources of data like databases and APIs.
+<!--more-->
+But we can also get data from any website even if it does not provide a public API. This process is known as **web scraping**, and it has been described in this article.
```
Comment (Dawe-7) [inline content/articles/introduction-to-web-scraping-with-javascript-and-puppeteer/index.md:18]:
```suggestion
But we can also get data from any website even if it does not provide a public API. This process is known as **web scraping**, and we will take a look at it in this article. 
```

File: content/articles/introduction-to-web-scraping-with-javascript-and-puppeteer/index.md [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+```
+
+The code is explained below:
+- `puppeteer.launch()` - It is a puppeteer method that launches the browser.
+- `puppeteer.newPage` - It is a puppeteer method used to open a new page.
+- `puppeteer.goto()` - It is a method that instructs the browser on which URL to go to. It takes two arguments, although the second one is optional. The first parameter is the actual URL. The second argument is used to disable timeout errors if the page you are loading is heavy.
+
+Next up, we need to grab the page markup and extract the exact information we need.
+
+Before doing this, open your browser and go to the url provided (https://www.amazon.com/Redragon-S101-Keyboard-Ergonomic-Programmable/dp/B00NLZUM36/). From this page, we need to get the image of the keyboard, its price, and its name. Follow these steps:
```
Comment (Dawe-7) [inline content/articles/introduction-to-web-scraping-with-javascript-and-puppeteer/index.md]:
```suggestion
Before doing this, open your browser and go to the [URL](https://www.amazon.com/Redragon-S101-Keyboard-Ergonomic-Programmable/dp/B00NLZUM36/)  provided. From this page, we need to get the image of the keyboard, its price, and its name. Follow these steps:
```

**Review by Dawe-7 — CHANGES_REQUESTED** [review]
Hello @kimanikevin254, awesome article, it's an interesting read and easy to follow through. I've left a few suggestions that i think will improve the quality of your article.  Also check on your article formatting, the headings should be `###` and subheadings should be `####` and lower. Lastly, adding more content in your conclusion will be nice.

> Ensure you pull the changes i've made before making the changes.

We will move on to tech review once you've implemented the changes.

**Comment by kimanikevin254** [comment]
@Dawe-7   Requested changes done

**Review by Dawe-7 — APPROVED** [review]
@kimanikevin254 , thanks for the changes.

@Kayere, this article is ready for final review.

**Comment by ahmadmardeni1** [comment]
Great work. Congrats on your first article and welcome to Section EngEd 🎉 @kimanikevin254 

Thank you for the review @Dawe-7 
Please go over the changes I made, and:
1. We always call it Node.js, not NodeJS.
2. Make sure to double-check the author's file in your next reviews.

**Pull request merged**